### ISSUE TITLE
Allow authoring native winrt api surface in C#

### ISSUE THREAD

**Issue by stevenbrix** [microsoft/CsWinRT#370]
The latest IDL syntax is very close to C#, but it still has limitations. Most notably, there is no syntax highlighting or intellisense. It would be great to just leverage the great C# toolchain that we have, including the ability to use [docfx](https://www.staticgen.com/docfx) to generate WinRT API information. The WinUI team has been authoring APIs in C# for years, and we have a custom tool that converts it to .idl. Authoring APIs in C# is nice, but having to maintain this tool is incredibly taxing.

Basically what this ask is, is that I can take a .NET ref assembly and run it through the new winmdexp tool and produce a .winmd.

**Comment by stevenbrix** [comment]
There are also limitations of midlrt, that i don't think are really a limitation of the winrt type system or anything, i.e. attributes. There just a limitation of the .idl format and toolchain

You can author an attribute like this:

```
attribute TypeHelperAttribute
{
   type HelperType;
   string ParseMethodName;
}
```
